### PR TITLE
[SYCL][Opaque pointer] Enable INTEL_SYCL_OPAQUEPOINTER_READY

### PR DESCRIPTION
**Pull request by jsji**
This is to enable -DINTEL_SYCL_OPAQUEPOINTER_READY=1 by default, 
so that we can start to remove the old code already removed by community (those guarded within #ifndef INTEL_SYCL_OPAQUEPOINTER_READY ).

- Set opauqe pointer ready
- Add flag to tablegen
- Remove opaque pointer from merge-extension-ivars.m
- Port barrier_intrinsic.ll to opaque pointer
- Remove intel-alloca-bitcast.ll as it is testing typed pointer path
- Revert changes to InferAddressSpaces/AMDGPU/icmp.ll
- Use private repo for testing purpose

### CODE REVIEW

**Comment by jsji** [comment]
Ping @intel/dpcpp-clang-driver-reviewers @intel/dpcpp-esimd-reviewers @intel/dpcpp-spirv-reviewers @intel/dpcpp-tools-reviewers . 

Most in these components are just test update, please have a look. Thanks.
The plan is to merge this before EOD tomorrow. Thanks.

**Review by sarnex — APPROVED** [review]
esimd lgtm

**Review by mdtoguchi — APPROVED** [review]

**Review by againull — APPROVED** [review]

File: llvm/lib/SYCLLowerIR/CMakeLists.txt [hunk 1]
```diff
@@ -12,11 +12,8 @@ endif()
 # NOTE: could have been added earlier from llvm/projects
 if (NOT TARGET LLVMGenXIntrinsics)
   if (NOT DEFINED LLVMGenXIntrinsics_SOURCE_DIR)
-    set(LLVMGenXIntrinsics_GIT_REPO https://github.com/intel/vc-intrinsics.git)
-    # Author: Gorban, Igor <igor.gorban@intel.com>
-    # Date:   Tue Jun 6 23:56:43 2023 +0000
-    # Mark llvm15 tests failures
-    set(LLVMGenXIntrinsics_GIT_TAG 910db4801d4a029834606e3e42a8d60358e74fdf)
+    set(LLVMGenXIntrinsics_GIT_REPO https://github.com/jsji/vc-intrinsics.git)
```
Comment (asudarsa) [inline llvm/lib/SYCLLowerIR/CMakeLists.txt:15]: I see the corresponding PR has been merged. Think this is ready for update. Just a reminder. Thanks

**Review by asudarsa — COMMENTED** [review]
Test changes in DPC++ Tools area looks good to me.
I do have a request for quick clarification for llvm-spirv changes here. Will it be possible to not make the changes directly here? We should be getting these changes as part of next pulldown from the Khronos SPIRV-LLVM-Translator repo. Adding @MrSidims and @jcranmer-intel for comment here.

Thanks a lot for putting up this major PR.

File: llvm/lib/SYCLLowerIR/CMakeLists.txt (hunk 1 above)
Comment (jsji) [reply llvm/lib/SYCLLowerIR/CMakeLists.txt:15]: Yes, but I am still waiting for the sync to open source repo. https://github.com/intel/vc-intrinsics

**Comment by jsji** [comment]
> Test changes in DPC++ Tools area looks good to me. I do have a request for quick clarification for llvm-spirv changes here. Will it be possible to not make the changes directly here? We should be getting these changes as part of next pulldown from the Khronos SPIRV-LLVM-Translator repo. Adding @MrSidims and @jcranmer-intel for comment here.
> 
> Thanks a lot for putting up this major PR.

Thanks for the review. The SPIRV changes are here because these are local to `sycl` branch, we have removed them in KhronosGroup/SPIRV-LLVM-Translator repo, but we added these options due to migration, this is to remove them so that we can in sync with KhronosGroup/SPIRV-LLVM-Translator repo again.

File: llvm/lib/SYCLLowerIR/CMakeLists.txt (hunk 1 above)
Comment (jsji) [reply llvm/lib/SYCLLowerIR/CMakeLists.txt]: Finally updated!

**Comment by jsji** [comment]
@asudarsa @MrSidims @jcranmer-intel Can you have a look? Hopefully we can get this merged today. :)

**Review by asudarsa — APPROVED** [review]
SPIR-V changes are in sync with open source. So we should be ok.

Thanks

**Comment by jsji** [comment]
Thanks @asudarsa ! @intel/llvm-gatekeepers Can we get this merged? Thank you!

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Can we get this merged? Thank you!

**Pull request merged**